Neos.Neos/Classes/Security/Authorization/ContentRepositoryAuthorizationService.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\Neos\Security\Authorization;
+
+use Neos\ContentRepository\Core\Feature\SubtreeTagging\Dto\SubtreeTags;
+use Neos\ContentRepository\Core\Projection\ContentGraph\Node;
+use Neos\ContentRepository\Core\Projection\ContentGraph\VisibilityConstraints;
+use Neos\ContentRepository\Core\SharedModel\ContentRepository\ContentRepositoryId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Security\Account;
+use Neos\Flow\Security\Authorization\PrivilegeManagerInterface;
+use Neos\Flow\Security\Policy\PolicyService;
+use Neos\Flow\Security\Policy\Role;
+use Neos\Neos\Domain\Model\NodePermissions;
+use Neos\Neos\Domain\Model\User;
+use Neos\Neos\Domain\Model\WorkspacePermissions;
+use Neos\Neos\Domain\Model\WorkspaceRole;
+use Neos\Neos\Domain\Model\WorkspaceRoleSubject;
+use Neos\Neos\Domain\Model\WorkspaceRoleSubjects;
+use Neos\Neos\Domain\Service\WorkspaceService;
+use Neos\Neos\Security\Authorization\Privilege\EditNodePrivilege;
+use Neos\Neos\Security\Authorization\Privilege\ReadNodePrivilege;
+use Neos\Neos\Security\Authorization\Privilege\SubtreeTagPrivilegeSubject;
+use Neos\Party\Domain\Service\PartyService;
+
+/**
+ * Central point which does ContentRepository authorization decisions within Neos.
+ *
+ * @api
+ */
+#[Flow\Scope('singleton')]
+final readonly class ContentRepositoryAuthorizationService
+{
+    private const FLOW_ROLE_EVERYBODY = 'Neos.Flow:Everybody';
+    private const FLOW_ROLE_ANONYMOUS = 'Neos.Flow:Anonymous';
+    private const FLOW_ROLE_AUTHENTICATED_USER = 'Neos.Flow:AuthenticatedUser';
+    private const FLOW_ROLE_NEOS_ADMINISTRATOR = 'Neos.Neos:Administrator';
+
+    public function __construct(
+        private PartyService $partyService,
+        private WorkspaceService $workspaceService,
+        private PolicyService $policyService,
+        private PrivilegeManagerInterface $privilegeManager,
+    ) {
+    }
+
+    /**
+     * Determines the {@see WorkspacePermissions} an anonymous user has for the specified workspace (aka "public access")
+     */
+    public function getWorkspacePermissionsForAnonymousUser(ContentRepositoryId $contentRepositoryId, WorkspaceName $workspaceName): WorkspacePermissions
```

Comment:
"Everybody" would be wrong IMO since it's _not_ everybody, it's the unauthenticated (aka anonymous) user.
To me "AnonymousUser" makes much more sense than "NoUser" in this context. But we can discuss that